config/core/300-resources/service.yaml

```diff
@@ -59,6 +59,1129 @@ spec:
+                type: object
+                description: "A template for the current desired application state.
+                  Changes to `template` will cause a new Revision to be created as
+                  defined in the lifecycle section. The contents of the Service's
+                  RevisionTemplateSpec is used to create a corresponding Configuration."
+                properties:
+                  metadata:
+                    type: object
+                    description: "Standard object's metadata."
+                    properties:
+                      annotations:
+                        type: object
+                        additionalProperties:
+                          type: string
+                        description: "Annotations is an unstructured key value map stored with
+                          a resource that may be set by external tools to store and retrieve
+                          arbitrary metadata. They are not queryable and should be preserved
+                          when modifying objects.
+                          \nMore info: http://kubernetes.io/docs/user-guide/annotations"
+                      clusterName:
+                        type: string
+                        description: "The name of the cluster which the object belongs to. This
+                          is used to distinguish resources with same name and namespace in different
+                          clusters. This field is not set anywhere right now and apiserver is
+                          going to ignore it if set in create or update request."
+                      creationTimestamp:
+                        format: date-time
+                        type: string
+                        description: "CreationTimestamp is a timestamp representing the server
+                          time when this object was created. It is not guaranteed to be set
+                          in happens-before order across separate operations. Clients may not
+                          set this value. It is represented in RFC3339 form and is in UTC.
+                          2020-10-28T15:00:00.05Z (YYYY-MM-DDTHH:MM:SS:MMZ)
+                          \nPopulated by the system.
+                          \nRead-only.
+                          \nNull for lists.
+                          \nMore info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata"
+                        nullable: true
+                      deletionGracePeriodSeconds:
+                        format: int64
+                        type: integer
+                        description: "Number of seconds allowed for this object to gracefully
+                          terminate before it will be removed from the system. Only set when
+                          deletionTimestamp is also set.
+                          \nMay only be shortened.
+                          \nRead-only."
+                      deletionTimestamp:
```

Comment:
There's a bunch of properties like `deletionTimestamp` that we don't want to accept in the template

Reply:
I think I found it here: https://github.com/knative/serving/blob/master/vendor/k8s.io/apimachinery/pkg/api/meta/meta.go#L131 

If it's not to be used should it be removed from here the API? Or should I not be using this file?

Reply:
I left this in waiting for further clarification.

Reply:
Is there a way to push this into the tooling?

That would simplify things since across the Knative project there are probably close to ~100 of CRDs

Reply:
seems like k8s rejects this over-specification

see the e2e integration checks where installation fails